code/datums/browser.dm

```diff
@@ -109,94 +119,47 @@
 	if(written_text)
 		. = user.handle_reading_literacy(user, .)
 
-/datum/browser/proc/open(var/use_onclose = 1)
+/datum/browser/proc/open(onclose = TRUE)
 	var/window_size = ""
-	if (width && height)
-		window_size = "size=[width]x[height];"
+	if(width && height)
+		window_size = "size=[width||0]x[height||0];"
+
 	show_browser(user, get_content(), "window=[window_id];[window_size][window_options]")
-	if (use_onclose)
+
+	if(onclose)
 		onclose(user, window_id, ref)
 
-/datum/browser/proc/update(var/force_open = 0, var/use_onclose = 1)
-	if(force_open)
-		open(use_onclose)
+/datum/browser/proc/update(force, onclose = TRUE)
+	if(force)
+		open(onclose)
 	else
 		send_output(user, get_content(), "[window_id].browser")
 
 /datum/browser/proc/close()
 	close_browser(user, "window=[window_id]")
 
-// This will allow you to show an icon in the browse window
-// This is added to mob so that it can be used without a reference to the browser object
-// There is probably a better place for this...
-/mob/proc/browse_rsc_icon(icon, icon_state, direction = -1)
-	/*
-	var/icon/I
-	if (direction >= 0)
-		I = new /icon(icon, icon_state, direction)
-	else
-		I = new /icon(icon, icon_state)
-		direction = "default"
-
-	var/filename = "[ckey("[icon]_[icon_state]_[direction]")].png"
-	send_rsc(src, I, filename)
-	return filename
-	*/
-
-
-// Registers the on-close verb for a browse window (client/verb/.windowclose)
-// this will be called when the close-button of a window is pressed.
-//
-// This is usually only needed for devices that regularly update the browse window,
-// e.g. canisters, timers, etc.
-//
-// windowid should be the specified window name
-// e.g. code is	: show_browser(user, text, "window=fred")
-// then use 	: onclose(user, "fred")
-//
-// Optionally, specify the "ref" parameter as the controlled atom (usually src)
-// to pass a "close=1" parameter to the atom's Topic() proc for special handling.
-// Otherwise, the user mob's machine var will be reset directly.
-//
-/proc/onclose(mob/user, windowid, var/atom/ref=null)
-	if(!user || !user.client) return
-	var/param = "null"
-	if(ref)
-		param = "\ref[ref]"
+/// Register windowclose to be called, optionally with ref, when the window window_id is closed
```

Comment:
Why are we removing the documentation here? This is a super non-intuitive feature, I'd rather all this stuff stayed.